http/AllowedHosts.cc

```diff
@@ -180,52 +143,40 @@ bool AllowedHosts::is_allowed(shared_ptr<http::url> candidate_url, std::string &
             try {
                 BESUtil::check_path(relative_path, catalog_root, follow_sym_links);
             }
-            catch (BESNotFoundError &e) {
-                why_not = e.get_message();
-                isAllowed = false;
-            }
-            catch (BESForbiddenError &e) {
+            catch (const BESError &e) {
                 why_not = e.get_message();
                 isAllowed = false;
             }
         }
-        BESDEBUG(MODULE, prolog << "File Access Allowed: " << (isAllowed ? "true " : "false ") << endl);
-    } else if(candidate_url->protocol() == HTTPS_PROTOCOL || candidate_url->protocol() == HTTP_PROTOCOL ){
-
-        isAllowed = candidate_url->is_trusted() || check(candidate_url->str());
+    }
+    else if (candidate_url.protocol() == HTTPS_PROTOCOL || candidate_url.protocol() == HTTP_PROTOCOL) {
 
-        BESDEBUG(MODULE, prolog << "HTTP Access Allowed: " << (isAllowed ? "true " : "false ") << endl);
+        isAllowed = candidate_url.is_trusted() || check(candidate_url.str());
+        why_not = "The URL is not on the AllowedHosts list and is not a known trusted URL";
     }
     else {
-        stringstream ss;
-        ss << "The candidate_url utilizes an unsupported protocol '" << candidate_url->protocol() << "'" ;
-        BESDEBUG(MODULE, prolog << ss.str() << endl);
-        throw BESInternalError(ss.str(),__FILE__,__LINE__);
+        why_not = "The URL utilizes an unsupported protocol:" + candidate_url.protocol();
+        isAllowed = false;
     }
-    BESDEBUG(MODULE, prolog << "END Access Allowed: " << (isAllowed ? "true " : "false ") << endl);
+
```

Comment:
I gotta say, I'm not keen on removing all these BESDEBUG calls from AllowedHosts (They cost nothing in the production version). I added them in here because this was a beast to debug...

Reply:
Added most of them back in.